File: samples/perl/theme.pl
use strict;
use warnings;
use feature 'say';

# Packages, hashes, references, regexes, interpolation, and substitutions.
package Theme;

sub new {
    my ($class, %args) = @_;
    return bless {
        name   => $args{name}   // 'CodePen Theme Original',
        accent => $args{accent} // '#96b38a',
        tokens => $args{tokens} // [],
    }, $class;
}

sub label {
    my ($self, $prefix) = @_;
    $prefix //= 'theme';
    return "$prefix: $self->{name} ($self->{accent})";
}

package main;

my $theme = Theme->new(tokens => [qw(keyword string comment)]);
my @visible = grep { $_ !~ /^comment$/ } @{$theme->{tokens}};
for my $index (0 .. $#visible) {
    my $token = uc $visible[$index];
    $token =~ s/STRING/LITERAL/;
    say sprintf '%d %s -> %s', $index + 1, $theme->label, $token;
}

File: samples/lua/theme.lua
-- Tables, metatables, methods, varargs, loops, and string interpolation.
local Theme = {}
Theme.__index = Theme

function Theme.new(name, accent, tokens)
  return setmetatable({
    name = name or "CodePen Theme Original",
    accent = accent or "#96b38a",
    tokens = tokens or {},
  }, Theme)
end

function Theme:label(prefix)
  return string.format("%s: %s (%s)", prefix or "theme", self.name, self.accent)
end

function Theme:visible_tokens(...)
  local hidden = { ... }
  local result = {}
  for index, token in ipairs(self.tokens) do
    if token ~= hidden[1] then
      result[index] = token:upper()
    end
  end
  return result
end

local original = Theme.new(nil, nil, { "keyword", "string", "comment" })
print(original:label(), table.concat(original:visible_tokens("comment"), ", "))
